src/components/autocomplete/js/autocompleteController.js

```diff
@@ -552,6 +561,30 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
   }
 
   /**
+   * Determines if the escape keydown should be processed
+   * @returns {boolean}
+   */
+  function shouldProcessEscape() {
+    return shouldBlurOnEscape() || !ctrl.hidden || ctrl.loading || shouldClearOnEscape() && $scope.searchText;
+  }
+
+  /**
+   * Determines if the escape key should blur focus
+   * @returns {boolean}
+   */
+  function shouldBlurOnEscape() {
+    return !$scope.escapeOptions || $scope.escapeOptions.toLowerCase().indexOf('blur') !== -1;
+  }
+
+  /**
+   * Determines if the escape key should clear input
+   * @returns {boolean}
+   */
+  function shouldClearOnEscape() {
+    return !$scope.escapeOptions || $scope.escapeOptions.toLowerCase().indexOf('clear') !== -1;
+  }
```

Comment:
Sorry for the nitpicky comments.

I'm just thinking that having those two functions is quite unnecessary.

```js
function hasEscapeOption(option) {
  return !$scope.escapeOptions || $scope.escapeOptions.toLowerCase().indexOf(option) !== -1;
}

// Then you can check the options like this
hasEscapeOption('clear');
```